netflix_3.0_word2vec.py

```diff
@@ -0,0 +1,155 @@
+import logging
+import sys
+import numpy
+from gensim.models.keyedvectors import KeyedVectors
+from math import log10
+
+from netflix_utils import NetflixUtils
+import netflix_data
+
+import tensorflow as tf
+
+BATCH_SIZE = 512
+DEFAULT_N_ITER = int(40 * netflix_data.NUM_RATING / BATCH_SIZE)
+REGULARIZATION_FACTOR = 0.5
+LEARNING_SPEED = 0.5
+MODEL_NAME = "netflix_0.0_latent"
+
+def preprocess_word_nt(movie_titles):
+    nt = {}
+    for movie_title in movie_titles:
+        for word in movie_title:
+            if word in nt:
+                nt[word] = nt[word] + 1
+            else:
+                nt[word] = 1
+    return nt
+
+def preprocess_title_embeddings(model, movie_titles, nt, ncol):
+    embeddings = numpy.zeros((len(movie_titles), ncol))
+    for i, movie_title in enumerate(movie_titles):
+        embedding = numpy.zeros((ncol))
+        length = 0
+        for word in movie_title:
+            if word in model.wv.vocab:
+                word_vector = model[word][0:ncol]
+                tfidf = log10(len(movie_titles) / nt[word])
+                embedding = embedding + tfidf * word_vector
+                length = length + 1
+        embedding = embedding / length
+        embeddings[i, :] = embedding
+
+    return embeddings
+
+def preprocess_get_titles(utils):
+    if utils.args.gen_movie_titles:
+        all_titles = []
```

Comment:
Same

Reply:
On ne connait pas le nombre de film à ce moment though. Est-ce que je devrais référer à la constante hard codé du nombre de film présente dans netflix_data?

Reply:
J'avoue qu'il faudrais ouvrir le fichier avant si on voulait savoir la longueur. La constante de netflix_data sounds good.

Reply:
Waiting for okay on that change and then merging PR